impl.go

```diff
@@ -305,6 +320,32 @@ func validReceiver(recv string) bool {
 	return err == nil
 }
 
+// commentsBefore returns true if comments precede the field. It returns false
+// otherwise.
+func commentsBefore(field *ast.Field, commentGroups []*ast.CommentGroup) bool {
+	if len(commentGroups) > 0 {
+		return commentGroups[0].Pos() < field.Pos()
+	}
+	return false
+}
+
+// flattenCommentMap flattens the comment map to a string.
+func flattenCommentMap(m ast.CommentMap) string {
+	var result strings.Builder
+	for _, cmmtgroups := range m {
+		for _, cmmtgroup := range cmmtgroups {
+			for _, cmt := range cmmtgroup.List {
+				result.WriteString(cmt.Text)
+				// add an end-of-line character if this is '//'-style comment
```

Comment:
It'd be nice for there to be a test including `/* ... */`-style comments.

Reply:
Similar suggestion here, if a test file with interfaces is introduced, one of the cases could be an interface with `/* ... */`-style comments on the methods.

Reply:
That definitely seems worth doing, regardless.

Reply:
I will add more commits later on with adequate testing for `/* ... */`-style comments.